File: backend/src/main/java/com/smartcampus/model/Resource.java
package com.smartcampus.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "resources")
public class Resource {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String type;

    @Column
    private Integer capacity;

    @Column
    private String location;

    @Column(nullable = false)
    private String status = "ACTIVE";

    @Column(length = 1000)
    private String description;
}

File: frontend/src/types/Resource.ts
export type ResourceStatus = 'ACTIVE' | 'OUT_OF_SERVICE';

export interface Resource {
  id: string;
  name: string;
  type: string;
  capacity: number;
  location: string;
  availabilityWindows: string[];
  status: ResourceStatus;
  metadata: Record<string, any>;
}
